sheetstorm/tests/test_unit_generate_test_data.py

```diff
@@ -327,3 +429,86 @@ def _compute_number_of_projects_to_create(options, projects_in_database, project
             raise UnsupportedProjectTypeException
 
         return number_of_projects_to_create
+
+
+class CreateDataFromPreparedSetTests(TestCase):
+    SUPERUSER_USER_TYPE = "SUPERUSER"
+
+    def setUp(self) -> None:
+        self.unrelated_options = {
+            CustomUser.UserType.ADMIN.name: 1,
+            CustomUser.UserType.EMPLOYEE.name: 1,
+            CustomUser.UserType.MANAGER.name: 1,
+            self.SUPERUSER_USER_TYPE: True,
+            ProjectType.SUSPENDED.name: 1,
+            ProjectType.ACTIVE.name: 1,
+            ProjectType.COMPLETED.name: 1,
+        }
+
+    def test_get_request_to_create_data_using_prepared_set_function_should_return_true_if_any_set_requested(self):
+        options = {
+            **self.unrelated_options,
+            DataSize.SMALL.name: True,
+            DataSize.MEDIUM.name: False,
+            DataSize.LARGE.name: False,
+            DataSize.EXTRA_LARGE.name: True,
+        }
+
+        self.assertTrue(GenerateTestDataCommand()._get_request_to_create_data_using_prepared_set(options))
+
+    def test_get_request_to_create_data_using_prepared_set_function_should_return_false_if_parameters_are_not_provided(
+        self
+    ):
+        options = {
+            **self.unrelated_options,
+            DataSize.SMALL.name: False,
+            DataSize.MEDIUM.name: False,
+            DataSize.LARGE.name: False,
+            DataSize.EXTRA_LARGE.name: False,
+        }
+
+        self.assertFalse(GenerateTestDataCommand()._get_request_to_create_data_using_prepared_set(options))
+
+    @parameterized.expand(
+        [
+            (True, False, False, False, DataSize.SMALL.name),
+            (False, True, False, False, DataSize.MEDIUM.name),
+            (False, False, True, False, DataSize.LARGE.name),
+            (False, False, False, True, DataSize.EXTRA_LARGE.name),
+        ]
+    )
+    def test_pick_dataset_to_create_function_should_return_one_specified_set_if_there_is_request(
+        self, small_requested, medium_requested, large_requested, extra_large_requested, set_name
+    ):
+        options = {
+            **self.unrelated_options,
+            DataSize.SMALL.name: small_requested,
+            DataSize.MEDIUM.name: medium_requested,
+            DataSize.LARGE.name: large_requested,
+            DataSize.EXTRA_LARGE.name: extra_large_requested,
+        }
+
+        self.assertEqual(GenerateTestDataCommand()._pick_dataset_to_create(options), DATA_SETS[set_name])
+
+    def test_pick_dataset_to_create_function_should_create_smaller_set_in_case_of_allow_to_specify_more_than_one(self):
+        options = {
+            **self.unrelated_options,
+            DataSize.SMALL.name: True,
+            DataSize.MEDIUM.name: False,
+            DataSize.LARGE.name: True,
+            DataSize.EXTRA_LARGE.name: True,
+        }
```

Comment:
Since it is currently impossible to specify more than one option, I don't think this should be tested, because right now it seems more like a side effect feature rather than a conscious design. 😉 

But that's more of my personal opinion, so I will leave the decisive voice to @kbeker